windows/src/developer/kmcmpdll/Compiler.cpp

```diff
@@ -1452,38 +1459,39 @@ DWORD AddCompilerVersionStore(PFILE_KEYBOARD fk)
 DWORD CheckStatementOffsets(PFILE_KEYBOARD fk, PFILE_GROUP gp, PWSTR context, PWSTR output, PWSTR key) {
   PWSTR p, q;
   int i;
-	for(p = output; *p; p=incxstr(p))	{
-		if(*p == UC_SENTINEL) {
-      if(*(p+1) == CODE_INDEX)	{
-        int indexStore = *(p+2) - 1;
-        int contextOffset = *(p+3);
-			  for(q = context, i = 1; *q && i < contextOffset; q=incxstr(q), i++);
-
-			  if(*q == 0)	{
-				  if(!gp->fUsingKeys)
+  for (p = output; *p; p = incxstr(p)) {
+    if (*p == UC_SENTINEL) {
+      if (*(p + 1) == CODE_INDEX) {
+        int indexStore = *(p + 2) - 1;
+        int contextOffset = *(p + 3);
+        for (q = context, i = 1; *q && i < contextOffset; q = incxstr(q), i++);
+
+        if (*q == 0) {
+          if (!gp->fUsingKeys)
             // no key in the rule, so offset is past end of context
             return CERR_IndexDoesNotPointToAny;
-				  if(i < contextOffset) // I4914
+          if (i < contextOffset) // I4914
             // offset is beyond the key
             return CERR_IndexDoesNotPointToAny;
-				  q = key;
-			  }
+          q = key;
+        }
 
-			  // find the any
-			  if(*q != UC_SENTINEL || *(q+1) != CODE_ANY)
+        // find the any
+        if (*q != UC_SENTINEL || *(q + 1) != CODE_ANY)
           return CERR_IndexDoesNotPointToAny;
 
-        int anyStore = *(q+2) - 1;
+        int anyStore = *(q + 2) - 1;
 
-			  if(xstrlen(fk->dpStoreArray[indexStore].dpString) < xstrlen(fk->dpStoreArray[anyStore].dpString))
-				  AddWarning(CWARN_IndexStoreShort);
-		  } else if(*(p+1) == CODE_CONTEXTEX) {
-        int contextOffset = *(p+2);
-        if(contextOffset > xstrlen(context))
+        if (xstrlen(fk->dpStoreArray[indexStore].dpString) < xstrlen(fk->dpStoreArray[anyStore].dpString))
+          AddWarning(CWARN_IndexStoreShort);
+      }
+      else if (*(p + 1) == CODE_CONTEXTEX) {
```

Comment:
Formatting here could use some work.

- the `if` is more indented than it should be.
- the `else` is not on the same line as the `if`'s closing brace.

Reply:
Yep; I am sorry about the reformatting that did occur -- I would have preferred to leave that to a separate PR. As it is, I can either leave the formatting as is, or restore it (except for pure whitespace changes) to previous. Which do you prefer?

Reply:
```suggestion
        if (xstrlen(fk->dpStoreArray[indexStore].dpString) < xstrlen(fk->dpStoreArray[anyStore].dpString)) {
          AddWarning(CWARN_IndexStoreShort);
        }
      } else if (*(p + 1) == CODE_CONTEXTEX) {
```